hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/codec/KeyValueWrapper.java

```diff
@@ -163,7 +163,8 @@ public Key asGrpc() {
             case DELEGATABLE_CONTRACT_ID -> Key.newBuilder()
                     .setDelegatableContractId(delegatableContractID)
                     .build();
-            default -> throw new InvalidTransactionException(ResponseCodeEnum.FAIL_INVALID);
+            default -> throw new InvalidTransactionException(
+                    "INVALID_KEY", ResponseCodeEnum.FAIL_INVALID);
```

Comment:
I think adding the detailed message "INVALID_KEY" is an improvement. Should we also change the response code to not be `FAIL_INVALID`? `FAIL_INVALID` is reserved for internal failures we're unable to recover from that may bring the whole node down. I think we can use something like `INVALID_TRANSACTION_BODY` here like we do in [verifySolidityInput()](https://github.com/hashgraph/hedera-services/blob/c107510646461121799ccefd036b465f0a809aee/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TokenCreatePrecompile.java#L472).